src/main/java/com/fasterxml/jackson/databind/ser/std/NumberSerializer.java

```diff
@@ -55,6 +64,9 @@ public JsonSerializer<?> createContextual(SerializerProvider prov,
         if (format != null) {
             switch (format.getShape()) {
             case STRING:
+                if (format.hasPattern()) {
+                    return new NumberSerializer(handledType(), new DecimalFormat(format.getPattern()));
```

Comment:
Probably needs a try-catch block for invalid format (and probably unit test to show handling), re-throw exception (using one of methods from `SerializerProvider`).